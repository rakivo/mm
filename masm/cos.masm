#COS x out {
    push x

    ; x^2
    dup 0
    dup 0
    fmul

    push 1.0

    ; x^2 / -2!
    dup 1
    push -2.0
    fdiv
    fadd

    ; x^4 / 4!
    dup 1
    dup 1
    fmul
    push 24.0
    fdiv
    fsub

    ; x^6 / 6!
    dup 1
    dup 1
    fmul
    push 720.0
    fdiv
    fadd

    ; x^8 / 8!
    dup 1
    dup 1
    fmul
    push 40320.0
    fdiv
    fsub

    ; x^10 / 10!
    dup 1
    dup 1
    fmul
    push 3628800.0
    fdiv
    fadd

    dmp out
}

#SYS_STDOUT 1
#ANGLE 0.6283185307179586

_start:
    COS ANGLE SYS_STDOUT
    halt
