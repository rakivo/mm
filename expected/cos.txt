0.7898307334362231

